spring-cloud-stream-test-support/src/main/java/org/springframework/cloud/stream/test/matcher/MessageQueueMatcher.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright 2015 the original author or authors.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.test.matcher;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.TimeUnit;
+
+import org.hamcrest.BaseMatcher;
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.hamcrest.SelfDescribing;
+
+import org.springframework.cloud.stream.test.binder.TestSupportBinder;
+import org.springframework.integration.util.Function;
+import org.springframework.messaging.Message;
+
+/**
+ * A Hamcrest Matcher meant to be used in conjunction with {@link TestSupportBinder}.
+ *
+ * <p>Expected usage is of the form (with appropriate static imports):
+ * <pre>
+ * public class TransformProcessorApplicationTests {
+ *
+ *    {@literal @}Autowired
+ *    {@literal @}ModuleChannels(TransformProcessor.class)
+ *    private Processor processor;
+ *
+ *    {@literal @}Autowired
+ *    private MessageCollectorImpl messageCollector;
+ *
+ *
+ *    {@literal @}Test
+ *    public void testUsingExpression() {
+ *        processor.input().send(new GenericMessage{@literal <}Object>("hello"));
+ *        assertThat(messageCollector.forChannel(processor.output()), receivesPayloadThat(is("hellofoo")).within(10));
+ *    }
+ *
+ * }</pre>
+ * </p>
+ *
+ * @author Eric Bottard
+ */
+public class MessageQueueMatcher<T> extends BaseMatcher<BlockingQueue<Message<?>>> {
```

Comment:
This class has a lot of warnings, such as "rawtypes" (showing up in the IDE). Also, is there any way to avoid some of the "unchecked" suppressions via parameterization?